chisel/src/executor.rs

```diff
@@ -132,17 +132,52 @@ impl SessionSource {
             Err(_) => return Ok((true, None)),
         };
 
-        // TODO: Any tuple fails compilation due to it not being able to be encoded in `inspectoor`
-        let mut res = match source.execute().await {
-            Ok((_, res)) => res,
-            Err(e) => {
-                if self.config.foundry_config.verbosity >= 3 {
-                    eprintln!("Could not inspect: {e}");
+        let mut source_without_inspector = self.clone();
+
+        // Events and tuples fails compilation due to it not being able to be encoded in `inspectoor`.
+        // If that happens, try executing without the inspector.
+        let (mut res, has_inspector) = match source.execute().await {
+            Ok((_, res)) => (res, true),
+            Err(e) => match source_without_inspector.execute().await {
+                Ok((_, res)) => (res, false),
+                Err(_) => {
+                    if self.config.foundry_config.verbosity >= 3 {
+                        eprintln!("Could not inspect: {e}");
+                    }
+                    return Ok((true, None));
                 }
-                return Ok((true, None))
-            }
+            },
         };
 
+        // If abi-encoding the input failed, check whether it is an event
+        if !has_inspector {
+            let generated_output = source_without_inspector
+                .generated_output
+                .as_ref()
+                .ok_or_else(|| eyre::eyre!("Could not find generated output!"))?;
+
+            let intermediate_contract = generated_output
+                .intermediate
+                .intermediate_contracts
+                .get("REPL")
+                .ok_or_else(|| eyre::eyre!("Could not find intermediate contract!"))?;
+
+            if let Some(event_definition) = intermediate_contract.event_definitions.get(input) {
+                let formatted = match format_event_definition(event_definition) {
+                    Ok(v) => Some(v),
+                    Err(err) => {
+                        if self.config.foundry_config.verbosity >= 3 {
+                            eprintln!("{err}");
+                        }
+                        None
+                    }
+                };
```

Comment:
This is an error we should return because 1. the normal parsing failed, 2. we know that it's an event and we failed parsing it

```suggestion
                let formatted = format_event_definition(event_definition)?;
                return Ok((false, Some(formatted)))
```